content/chainguard/chainguard-images/chainguard-registry/pull-through-guides/harbor/index.md

```diff
@@ -0,0 +1,132 @@
+
+## Prerequisites
+
+You need the following in order to complete this tutorial:
+
+* Administrative privileges over a Harbor instance. Refer to the [official Harbor documentation](https://goharbor.io/docs/2.13.0/) to learn how to set this up.
+* `chainctl` — Chainguard's command-line interface — installed on your local machine. If you don't have `chainctl` installed, refer to our [How to Install `chainctl`](/chainguard/chainctl-usage/how-to-install-chainctl/) guide to set this up.
+* Permissions pull Chainguard container images from your organization's repository within the Chainguard registry. This is necessary, as you will create a pull token for Harbor to use to access the registry, and you cannot generate a pull token that grants broader access than your own.
+
+
+## Create a Registry Endpoint
+
+Before configuring a proxy cache or replication rule, you must create a registry endpoint for the Chainguard registry.
+
+If you don't already have one, generate a pull token in your organization:
+
+```shell
+chainctl auth configure-docker --parent <org-name> --pull-token
+```
+
+This returns username and password credentials:
+
+```Output
+To use this pull token in another environment, run this command:
+
+    docker login "cgr.dev" --username "<pull-token-username>" --password "<pull-token-password>"
+```
+
+Note these values down, as you'll need them shortly.
+
+Next, open up the Harbor UI and perform the following steps:
+
+1. Navigate to **Administration > Registries**
+2. Click the **+ NEW ENDPOINT** button..
+3. Configure the endpoint with these details:
+    * **Provider** — There is no specific integration with the Chainguard registry, so here you must select the generic `Docker Registry` provider type.
+    * **Name** — This is used to refer to your repository. You can choose whatever name you like here, but this guide's examples use the name `cgr.dev`.
+    * **Endpoint URL** — This value **must** be `https://cgr.dev`.
+    * **Access ID** — Enter the username value returned by the `chainctl auth` command you just ran.
+    * **Access Secret** — Here, enter the password value returned by the `chainctl auth` command.
+4. Click the **TEST CONNECTION** button to ensure that Harbor can reach `cgr.dev` successfully.
+5. Click **OK** to create the endpoint.
+
+After creating the endpoint, you can move on to creating a proxy cache.
+
+
+## Create a Proxy Cache
+
+A [*proxy* cache](https://goharbor.io/docs/2.1.0/administration/configure-proxy-cache/) allows Harbor to proxy and cache images from the Chainguard registry.
+
+To configure a cache, perform the following steps in the Harbor UI:
+
+1. Navigate to **Projects**.
+2. Click the **+ NEW PROJECT** button.
+3. Create the project with these details:
+    * **Project Name** — This can be whatever you like. This guide uses the name `cgr—proxy`.
```

Comment:
Add "Access Level -- set as required for your organization. Setting to public means docker login is not required"

Reply:
Add "Project quota limits -- leave at -1 for unlimited or set as required"